File: src/main/java/org/virajshah/monopoly/tiles/BasicTile.java
package org.virajshah.monopoly.tiles;

import java.util.List;

public class BasicTile extends Tile {
	/**
	 * @param name       The name of the tile
	 * @param attributes Attributes associated with the tile
	 */
	public BasicTile(String name, List<TileAttribute> attributes) {
		super(name, attributes);
	}

	/**
	 * @param name       The name of the tile
	 * @param attributes Attributes associated with the tile
	 */
	public BasicTile(String name, TileAttribute[] attributes) {
		super(name, attributes);
	}

	/**
	 * @param name       The name of the tile
	 * @param attributes Attributes associated with the tile
	 */
	public BasicTile(String name, TileAttribute attribute) {
		super(name, attribute);
	}
}

File: src/main/java/org/virajshah/monopoly/tiles/BasicTile.kt
package org.virajshah.monopoly.tiles

class BasicTile : Tile {
    constructor(name: String, attributes: List<TileAttribute?>) : super(name, attributes)

    constructor(name: String, attributes: Array<TileAttribute?>) : super(name, attributes)

    constructor(name: String, attribute: TileAttribute?) : super(name, attribute)
}